### DISCUSSION TITLE
Full Authentication Required

### DISCUSSION THREAD

**Discussion by hashamrasheed** [flamingock/mongock#55] [Q&A] [answered]
Hello. I'm new here, so forgive me if I don't get this right. I'm using Mongock with Spring Boot 2.1.10 with mongock-version=4.1.19,mongodb-java-driver-version=3.11.0 and spring-data-mongodb-version=2.1.8.RELEASE . Below is my pom.xml:

`      

        <dependency>
            <groupId>com.github.cloudyrock.mongock</groupId>
            <artifactId>mongock-bom</artifactId>
            <version>${mongock-version}</version>
            <type>pom</type>
            <scope>import</scope>
        </dependency>

        <dependency>
            <groupId>com.github.cloudyrock.mongock</groupId>
            <artifactId>mongock-spring-v5</artifactId>
            <version>${mongock-version}</version>
        </dependency>

        <dependency>
            <groupId>com.github.cloudyrock.mongock</groupId>
            <artifactId>mongodb-springdata-v2-driver</artifactId>
            <version>${mongock-version}</version>
        </dependency>

        <dependency>
            <groupId>org.mongodb</groupId>
            <artifactId>mongo-java-driver</artifactId>
            <version>${mongodb-java-driver-version}</version>
        </dependency>

        <dependency>
            <groupId>org.springframework.data</groupId>
            <artifactId>spring-data-mongodb</artifactId>
            <version>${spring-data-mongodb-version}</version>
        </dependency>
`

I'm trying to delete a user using the following changeSet:

`  @ChangeSet(order = "000001", id = "deleteUser", author = "test")
  public void deleteUser(MongoDatabase db) {
    LOG.info("Deleting First");
    db.runCommand(new BasicDBObject("dropUser", "testUser"));
  }`

I get the following error in console:
`
Error in changock process. ABORTED MIGRATION
io.changock.migration.api.exception.ChangockException: Error in method[DatabaseChangelogs.deleteUser] : Command failed with error 13 (Unauthorized): 'not authorized on testDB to execute command { dropUser: "testUser", $db: "testDB" }' on server mongo:27017. The full response is { "ok" : 0.0, "errmsg" : "not authorized on testDB to execute command { dropUser: \"testUser\", $db: \"testDB\" }", "code" : 13, "codeName" : "Unauthorized" }
`

**Comment by dieppa** [comment, marked as answer]
Hello,

as long as I can see that's not an issue with Mongock, but MongoDB itself. It seems you are trying to delete a user from the database, but you require the right authentication as that's an admin operation.

Please check you get the same error when doing the same out of Mongock changelog.

If you want to do that, as I believe you don't won't your application to work with a MongoTemplate with admin rights, I suggest you to create a specific mongoTemplate  for Mongock with the right connection string(or authenticate it somehow) and then inject it to Mogock via builder approach. The @EnableMongock would pick the primary bean and, as said, you don't want the primary MongoTemplate to have admin rights.

**Reply by hashamrasheed** [reply]
Thanks @dieppa . I do see a document in mongockChangeLock, but I don't particularly see any message.
I actually want my changeLogs to have admin rights. Is there a way I can do this without having to use builder approach?

**Reply by dieppa** [reply]
You are probably seeing a failed changeLog in the collection mongockChangeLog. That's normal as the failed changeLogs are tracked to but they are re-tried until they are successful.

Currently there is no other way. As part of the version 5, we are providing a feature that allows you to select the MongoTemplate you wan to be injected to Mongock, so you don't need to use the builder.
